server/src/main/java/org/elasticsearch/common/time/DateFormatter.java

```diff
@@ -129,19 +129,17 @@ static DateFormatter forPattern(String input) {
         if (Strings.hasLength(input) == false) {
             throw new IllegalArgumentException("No date pattern provided");
         }
+        if (input.startsWith("8") == false) {
+            return Joda.forPattern(input);
+        }
+
+        // force java 8 date format
         List<DateFormatter> formatters = new ArrayList<>();
         for (String pattern : Strings.delimitedListToStringArray(input, "||")) {
             if (Strings.hasLength(input) == false) {
                 throw new IllegalArgumentException("Cannot have empty element in multi date format pattern: " + input);
             }
-            final DateFormatter formatter;
-            if (pattern.startsWith("8")) {
-                // force java 8 date format
-                formatter = DateFormatters.forPattern(pattern.substring(1));
-            } else {
-                formatter = Joda.forPattern(pattern);
-            }
-            formatters.add(formatter);
+            formatters.add(DateFormatters.forPattern(pattern.substring(1)));
```

Comment:
this needs a check for each of the pattern if it starts with `8` in order to return the proper joda time pattern for sth like `8date_optional_time||date_optional_time`

Reply:
Oops, this is leftover. We won't be able to have mixed joda/java formats, so I think only having `8` as the very first character is sufficient (otherwise it would be redundant). I'll move the substring to outside the loop.